### PR TITLE
Build `nix-store` with Meson, adjust Perl bindings deps to integrate with other meson builds

### PR DESCRIPTION
**Pull request by Ericson2314**
# Motivation

More progress on #2503 

# Context

Following up to #10855

# Priorities and Process

Add :+1: to [pull requests you find important](https://github.com/NixOS/nix/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc).

The Nix maintainer team uses a [GitHub project board](https://github.com/orgs/NixOS/projects/19) to [schedule and track reviews](https://github.com/NixOS/nix/tree/master/maintainers#project-board-protocol).

### CODE REVIEW

**Comment by Ericson2314** [comment]
@eli-schwartz Any idea why it might be hanging in the generate steps? It didn't happen for me doing similar cross builds locally.

**Comment by eli-schwartz** [comment]
I'm looking and I'm seeing a CI error, but it doesn't have anything to do with generating SQL.

> The hosted runner: GitHub Actions 5 lost communication with the server. Anything in your workflow that terminates the runner process, starves it for CPU/Memory, or blocks its network access can cause this error.

**Comment by eli-schwartz** [comment]
It says it failed in 47 minutes but the job stages with logs are only a couple seconds long total. This is unquestionably a GitHub-level bug in GitHub's own infrastructure, because something broke on their end mid-run.

Restarting probably helps.

**Comment by Ericson2314** [comment]
@eli-schwartz Sorry. Yes there is a bug that makes the logs go away. Very frustrating. But I did restart it with the logs kept open after a few times of this to see that it was consistently stalling when building the SQL-processing steps.

**Comment by Ericson2314** [comment]
(https://github.com/NixOS/nix/pull/10855 was approved with this a part of it. I pulled it out just for purposes of debugging CI failures.)

**Comment by Ericson2314** [comment]
(I went to restart a timed out macOS job, and then that briefly caused all checks to pass, which resulted in an auto merge. Yay GitHub Action...)

File: src/libstore/meson.build [hunk 1]
```diff
@@ -0,0 +1,452 @@
...
+
+# See note in ../nix-util/meson.build
+deps_other = [ ]
+
+configdata = configuration_data()
+
+# TODO rename, because it will conflict with downstream projects
+configdata.set_quoted('PACKAGE_VERSION', meson.project_version())
+
+configdata.set_quoted('SYSTEM', host_machine.system())
```
Comment (roberth) [inline src/libstore/meson.build:31]: This is probably wrong. `host_machine.system()` is expected to have e.g. `'linux'` in other places.

I don't know how to get the right value, reliably, and we've [defined it to be based on autoconf](https://nix.dev/manual/nix/2.22/contributing/hacking.html?highlight=system#system-type), which I don't think we want to continue doing, or do we? Alternatively, we could leave it to the packager, and pass `stdenv.hostPlatform.system` in our own packaging?

**Pull request merged**